Zend/zend_builtin_functions.c

```diff
@@ -1557,27 +1559,77 @@ static void debug_backtrace_get_args(zend_execute_data *call, zval *arg_array) /
 					 */
 					while (i < first_extra_arg) {
 						zend_string *arg_name = call->func->op_array.vars[i];
-						zval *arg = zend_hash_find_ex_ind(call->symbol_table, arg_name, 1);
-						if (arg) {
-							ZVAL_DEREF(arg);
-							Z_TRY_ADDREF_P(arg);
-							ZEND_HASH_FILL_SET(arg);
+						zval arg;
+						zval *old_arg = zend_hash_find_ex_ind(call->symbol_table, arg_name, 1);
+						zend_attribute *attribute = zend_get_parameter_attribute_str(
+							call->func->op_array.attributes,
+							"sensitiveparameter",
+							sizeof("sensitiveparameter") - 1,
+							i
+						);
+
+						if (attribute != NULL) {
+							last_was_sensitive = 1;
 						} else {
-							ZEND_HASH_FILL_SET_NULL();
+							last_was_sensitive = 0;
+							
 						}
+
+						if (old_arg) {
+							ZVAL_DEREF(old_arg);
+							Z_TRY_ADDREF_P(old_arg);
+							ZVAL_COPY_VALUE(&arg, old_arg);
```

Comment:
Yes, this appears to pass all tests. I was hoping that it would also fix https://github.com/php/php-src/pull/7921#issuecomment-1035987956, but unfortunately it does not.

Reply:
I think that's due to a double `Z_TRY_ADDREF_P` here and in `ZEND_METHOD(SensitiveParameterValue, __construct)`. Try reverting my suggestion and move the `Z_TRY_ADDREF_P(&arg)` to the `else` case of `if (last_was_sensitive) {` below.

Reply:
Okay, so as I've assumed. Thanks!